hstio/hstio.c

```diff
@@ -277,7 +277,7 @@ int findTotalNumberOfHDUSets(const char * fileName, const char * setContainsExtN
             int match = FALSE;
             int exact = FALSE;
             int caseSensitive = TRUE;
-            fits_compare_str(setContainsExtName, keyValue, caseSensitive, &match, &exact);
+            fits_compare_str((char *) setContainsExtName, keyValue, caseSensitive, &match, &exact);
```

Comment:
`findTotalNumberOfHDUSets` expects `setContainsExtN` to be `const char *`, but `fits_compare_str` wants a `char *`. As far as I can see, no modifications to the pointer are made in `fits_compare_str`. This just squelches the warning.